docs/containers/reference.md

```diff
@@ -392,6 +392,74 @@ Here are all properties available for configuring `docker-run` task. All propert
+    "tasks": [
+        {
+            "label": "Run docker-compose up",
+            "type": "docker-compose",
+            "up": {
+                "detached": true,
+                "build": true,
+                "services": [
+                  "myservice"
+                ]
+            },
+            "files": [
+                "${workspaceFolder}/docker-compose.yml",
+                "${workspaceFolder}/docker-compose.debug.yml"
+            ]
+        }
+    ]
+}
+```
+
+## Compose task reference
+
+Here are all properties available for configuring `docker-compose` task. All properties are optional unless indicated otherwise.
+
+| Property | Description |
+| --- | --- |
+| `dockerCompose` | Options for controlling the `docker-compose` command executed ([see below](#dockercompose-object-properties)). <br/> Required. |
+
+### dockerCompose object properties
+
+| Property | Description | CLI Equivalent |
+| --- | --- | --- |
+| `up` | Run a `docker-compose up` command. <br/> Either this or `down` must be specified, but not both. | `docker-compose up` |
+| `down` | Run a `docker-compose down` command. <br/> Either this or `up` must be specified, but not both. | `docker-compose down` |
+| `files` | The list of Docker Compose YAML files to use in the `docker-compose` command. If not specified, the Docker Compose CLI looks for `docker-compose.yml` and `docker-compose.override.yml`. | `-f <file>` |
+
+### up object properties
+
+| Property | Description | CLI Equivalent | Default |
+| --- | --- | --- | --- |
+| `detached` | Whether or not to run detached. | `-d` | `true` |
+| `build` | Whether or not to build before running. | `--build` | `true` |
+| `scale` | Number of instances of each service to run. This is a list of key-value pairs. | `--scale SERVICE=NUM` |
+| `services` | A subset of the services to start. | `[SERVICE...]` | (all) |
+| `customOptions` | Any extra parameters to add after the `up` argument. No attempt is made to resolve conflicts with other options or validate this option. | (any) |
+
+### down object properties
+
+| Property | Description | CLI Equivalent | Default |
+| --- | --- | --- | --- |
+| `removeImages` | Whether to remove images, and which. `all` will remove all images used by any service, `local` will remove only images without a custom tag. Leaving this unset will remove no images. | `--rmi` |
+| `removeVolumes` | Whether or not to remove named volumes. | `-v` | `false` |
+| `customOptions` | Any extra parameters to add after the `down` argument. No attempt is made to resolve conflicts with other options or validate this option. | (any) |
+
 ## Command customization
 
 The Docker extension executes a number of Docker CLI commands when you perform various operations, such as to build images, run containers, attach to containers, and view container logs. Some of these commands have a large number of optional arguments, often used in very specific scenarios. Many of these commands can be customized.
```

Comment:
@BigMorty suggested adding some prose up here to draw extra attention to the `${serviceList}` token.